### PR TITLE
Escapse the repository name in the link header returned in response

### PR DESCRIPTION
**Pull request by ywk253100**
Escapse the repository name in the link header returned in response

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: src/server/v2.0/handler/base.go [hunk 1]
```diff
@@ -121,26 +121,44 @@ func (b *BaseAPI) BuildQuery(ctx context.Context, query *string, pageNumber, pag
...
 	var links internal.Links
 	// prev
 	if pageNumber > 1 && (pageNumber-1)*pageSize < total {
-		q := url.Query()
+		q := ul.Query()
 		q.Set("page", strconv.FormatInt(pageNumber-1, 10))
-		url.RawQuery = q.Encode()
+		ul.RawQuery = q.Encode()
+		// try to unescape the query
+		if escapedQuery, err := url.QueryUnescape(ul.RawQuery); err == nil {
```
Comment (heww) [inline src/server/v2.0/handler/base.go]: Why do we need to unescape query string?

Comment (ywk253100) [reply src/server/v2.0/handler/base.go]: pkg/q/builder.Build() can only handle the unescaped characters

Comment (heww) [reply src/server/v2.0/handler/base.go]: I think the `q` is unescaped automic before pass it the `pkg/q/builder.Build`, is it right?

And we need to response the same values as they are in the query string, we only change the `page` paramater.

Comment (ywk253100) [reply src/server/v2.0/handler/base.go]: In line 136, we encode the query string, the special characters used in our query(~,[],etc) are all encoded, so here decode them first

**Pull request merged**